src/lib.rs

```diff
@@ -226,6 +247,71 @@ pub struct PointsBatch {
 }
 
 impl PointsBatch {
+    pub fn append(&mut self, other: &mut PointsBatch) -> std::result::Result<(), String> {
+        if self.position.is_empty() {
+            *self = other.split_off(0);
+        } else {
+            assert_eq!(self.attributes.len(), other.attributes.len());
+            self.position.append(&mut other.position);
+            let data_types: Vec<(String, AttributeDataType)> = self
+                .attributes
+                .iter()
+                .map(|(name, attrib)| (name.clone(), attrib.data_type()))
+                .collect();
+            for (name, dtype) in &data_types {
+                use AttributeDataType::*;
+                match dtype {
+                    U8 => append_attribute::<u8>(name, self, other)?,
+                    U64 => append_attribute::<u64>(name, self, other)?,
+                    I64 => append_attribute::<i64>(name, self, other)?,
+                    F32 => append_attribute::<f32>(name, self, other)?,
+                    F64 => append_attribute::<f64>(name, self, other)?,
+                    U8Vec3 => append_attribute::<Vector3<u8>>(name, self, other)?,
+                    F64Vec3 => append_attribute::<Vector3<f64>>(name, self, other)?,
+                };
+            }
+        }
+        Ok(())
+    }
+
+    pub fn split_off(&mut self, at: usize) -> Self {
+        let mut res = Self {
+            position: self.position.split_off(at),
```

Comment:
This can panic – is that acceptable?

Reply:
I tried to mimic the same behavior as the corresponding `Vec` functions. For me it is acceptable, yes.